src/RelayClient.ts

```diff
@@ -286,160 +316,133 @@ class RelayClient extends EnvelopingEventEmitter {
-      // is required to estimate the token gas. This value should be calculated prior to
-      // the call to this function
-      if (!tokenOrigin || tokenOrigin === constants.AddressZero) {
-        throw Error(MISSING_SMART_WALLET_ADDRESS);
-      }
-    } else {
-      tokenOrigin = callForwarder.toString();
-
-      if (tokenOrigin === constants.AddressZero) {
-        throw Error(MISSING_CALL_FORWARDER);
-      }
-    }
-
-    const erc20 = IERC20__factory.connect(
-      tokenContract.toString(),
-      this._provider
-    );
-    const gasCost = await erc20.estimateGas.transfer(
-      feesReceiver,
-      tokenAmount,
-      { from: tokenOrigin, gasPrice }
+    const envelopingTx = await this._prepareHttpRequest(
+      activeRelay!.hubInfo,
+      envelopingRequestDetails
     );
 
-    const internalCallCost = applyInternalEstimationCorrection(
-      gasCost,
-      internalEstimationCorrection
-    );
+    await this._verifyEnvelopingRequest(activeRelay!.hubInfo, envelopingTx)
 
-    return applyGasCorrectionFactor(
-      internalCallCost,
-      estimatedGasCorrectionFactor
+    const transaction = await this._attemptRelayTransaction(
+      activeRelay!,
+      envelopingTx
     );
-  }
 
-  public async relayTransaction(
-    envelopingRequest: UserDefinedEnvelopingRequest,
-    requestConfig: RequestConfig
-  ): Promise<Transaction> {
-    const envelopingRequestDetails = await this._getEnvelopingRequestDetails(envelopingRequest, requestConfig);
+    if (transaction) {
+      log.debug('Relay Client - Relayed done');
 
-    log.debug('Relay Client - Relaying transaction');
-    log.debug(`Relay Client - Relay Hub:${envelopingRequestDetails.request.relayHub.toString()}`);
+      return transaction;
+    }
 
-    let activeRelay = await selectNextRelay(this._httpClient);
 
-    while (activeRelay) {
-      const envelopingTx = await this._prepareHttpRequest(activeRelay.hubInfo, envelopingRequestDetails, requestConfig);
+    throw Error(NOT_RELAYED_TRANSACTION);
 
-      if (await this._verifyEnvelopingRequest(activeRelay.hubInfo, envelopingTx)) {
+  }
 
-        const transaction = await this._attemptRelayTransaction(activeRelay, envelopingTx);
+  public async estimateRelayTransaction(
+    envelopingRequest: UserDefinedEnvelopingRequest
+  ) {
+    const envelopingRequestDetails = await this._getEnvelopingRequestDetails(
+      envelopingRequest
+    );
 
-        if (transaction) {
-          log.debug('Relay Client - Relayed done');
+    log.debug('Relay Client - Estimating transaction');
+    log.debug(
+      `Relay Client - Relay Hub:${envelopingRequestDetails.request.relayHub.toString()}`
+    );
 
-          return transaction;
-        }
-      }
+    const activeRelay = await selectNextRelay(this._httpClient);
 
-      activeRelay = await selectNextRelay(this._httpClient);
-    }
+    const envelopingTx = await this._prepareHttpRequest(
+      activeRelay!.hubInfo,
+      envelopingRequestDetails
+    );
```

Comment:
Unless I'm missing something, it looks like this part is identical to what we have in the relayTransaction (L#365-376)

Reply:
now is the same, but before it was different.

Reply:
updated